File: c++/reverseString.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int st = 0;
        int e = s.size() - 1;
        while(st < e) {
            swap(s[st++], s[e--]);
        }
    }
};

int main() {
    // Example input string as vector<char>
    vector<char> s = {'h', 'e', 'l', 'l', 'o'};

    Solution sol;
    sol.reverseString(s);

    // Printing the reversed string
    for(char c : s) {
        cout << c;
    }
    cout << endl;

    return 0;
}

File: javascript/reverseString.js
// Filename: reverseString.js

// Function to reverse a string
function reverseString(str) {
    // Step 1: Convert string into an array
    // Step 2: Reverse the array
    // Step 3: Join the array back into a string
    return str.split('').reverse().join('');
  }
  
  // Test the function
  const input = "hello";
  const result = reverseString(input);
  
  console.log("Input:", input);
  console.log("Reversed:", result);
  